### PR TITLE
FMWK-460 Add example for reactive flow

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

**Review by reugn — APPROVED** [review]

File: spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeConsumersConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        aerospikeTemplateRef = "reactiveAerospikeTemplateConsumers")
+public class AerospikeConsumersConfiguration extends AbstractReactiveAerospikeDataConfiguration {
+    @Override
+    protected Collection<Host> getHosts() {
+        return Collections.singleton(new Host("localhost", 3000));
+    }
+
+    @Override
+    protected String nameSpace() {
+        return "users";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeConsumersConfiguration.java]: should be "consumers"

File: spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeConsumersConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    }
+
+    @Override
+    protected ClientPolicy getClientPolicy() {
+        ClientPolicy clientPolicy = super.getClientPolicy(); // applying default values first
+        clientPolicy.user = "tester";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeConsumersConfiguration.java]: Change credentials

File: spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeItemsConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+    @Override
+    protected String nameSpace() {
+        return "items";
+    }
+
+    @Override
+    protected ClientPolicy getClientPolicy() {
+        ClientPolicy clientPolicy = super.getClientPolicy(); // applying default values first
+        clientPolicy.user = "tester";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/configuration/AerospikeItemsConfiguration.java]: Change credentials

File: spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Autowired
+    ReactiveAerospikeConsumersRepository aerospikeUsersReactiveRepository;
+    @Autowired
+    ReactiveAerospikeItemsRepository aerospikeItemsReactiveRepository;
+
+    @Test
+    public void consumerTest() {
+        Consumer consumer = new Consumer(1, "consumer1", "consumer1@gmail.com", 30);
+        aerospikeUsersReactiveRepository.save(consumer).block();
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java]: use aerospikeConsumerReactiveRepository, not users repository

File: spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+public class ExampleReactiveTests {
+
+    @Autowired
+    ReactiveAerospikeConsumersRepository aerospikeUsersReactiveRepository;
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java]: Name of the repo is wrong, should be consumers and not users

File: spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    @Autowired
+    ReactiveAerospikeItemsRepository aerospikeItemsReactiveRepository;
+
+    @Test
+    public void consumerTest() {
+        Consumer consumer = new Consumer(1, "consumer1", "consumer1@gmail.com", 30);
+        aerospikeUsersReactiveRepository.save(consumer).block();
+
+        Optional<Consumer> userFromDB = aerospikeUsersReactiveRepository.findById(consumer.getId()).blockOptional();
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java]: Also change the name of the param to `consumerFromDB`

File: spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+        assertThat(userFromDB).hasValue(consumer);
+    }
+
+    @Test
+    public void itemTest() {
+        Item item = new Item(1, "item1", "item1@gmail.com", "iron");
+        aerospikeItemsReactiveRepository.save(item).block();
+
+        Optional<Item> productFromDB = aerospikeItemsReactiveRepository.findById(item.getId()).blockOptional();
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-reactive/src/test/java/com/aerospike/demo/springdatamultiplenamespacesexamplereactive/ExampleReactiveTests.java]: `itemFromDB`

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeProductsConfiguration.java [hunk 8]
```diff
@@ -36,6 +37,14 @@ protected String nameSpace() {
         return "products";
     }
 
+    @Override
+    protected ClientPolicy getClientPolicy() {
+        ClientPolicy clientPolicy = super.getClientPolicy(); // applying default values first
+        clientPolicy.user = "tester";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeProductsConfiguration.java]: change credentials

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java [hunk 9]
```diff
@@ -24,20 +25,28 @@
 @Configuration
 @EnableAerospikeRepositories(basePackageClasses = AerospikeUsersRepository.class,
         includeFilters = @ComponentScan.Filter(UsersRepository.class),
-        aerospikeTemplateRef = "aerospikeTemplateUsers")
-public class AerospikeUsersConfiguration extends AbstractAerospikeDataConfiguration {
+        aerospikeTemplateRef = "aerospikeTemplateConsumers")
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java]: Should be users and not consumers (in the sync flow)

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java [hunk 10]
```diff
@@ -24,20 +25,28 @@
...
+public class AerospikeUserConfiguration extends AbstractAerospikeDataConfiguration {
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java]: No need to change, keep as it was (plural also aligns with other classes)

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java [hunk 11]
```diff
@@ -24,20 +25,28 @@
...
     @Override
     protected Collection<Host> getHosts() {
         return Collections.singleton(new Host("localhost", 3000));
     }
 
     @Override
     protected String nameSpace() {
-        return "users";
+        return "consumers";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java]: should be "users"

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java [hunk 12]
```diff
@@ -24,20 +25,28 @@
...
     }
 
-    @Bean(name = "aerospikeTemplateUsers")
-    public AerospikeTemplate aerospikeTemplateUsers(IAerospikeClient aerospikeClient,
+    @Override
+    protected ClientPolicy getClientPolicy() {
+        ClientPolicy clientPolicy = super.getClientPolicy(); // applying default values first
+        clientPolicy.user = "tester";
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java]: change credentials

File: spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java [hunk 13]
```diff
@@ -24,20 +25,28 @@
...
+        clientPolicy.password = "psw";
+        return clientPolicy;
+    }
+
+    @Bean(name = "aerospikeTemplateConsumers")
```
Comment (roimenashe) [inline spring-data-multiple-namespaces-example-sync/src/main/java/com/aerospike/demo/springdatamultiplenamespacesexample/configuration/AerospikeUserConfiguration.java]: Method and bean name should be users

**Review by roimenashe — APPROVED** [review]

**Pull request merged**